docs/fsharp/language-reference/generics/constraints.md

```diff
@@ -21,6 +21,7 @@ There are several different constraints you can apply to limit the types that ca
 |----------|------|-----------|
 |Type Constraint|*type-parameter* :&gt; *type*|The provided type must be equal to or derived from the type specified, or, if the type is an interface, the provided type must implement the interface.|
 |Null Constraint|*type-parameter* : null|The provided type must support the null literal. This includes all .NET object types but not F# list, tuple, function, class, record, or union types.|
+|Not Null Constraint|*type-parameter* : not null|The provided type must not support the null literal. This disallows both `null` annotated types and types which have null as their representation value (such as the option type). This does allow value types, since those can never be null.|
```

Comment:
```suggestion
|Not Null Constraint|*type-parameter* : not null|The provided type must not support the null value. This disallows both `null` annotated types and types which have null as their representation value (such as the option type or types defined with AllowNullLiteral attribute). This generic constraint does allow value types, since those can never be null.|
```

Reply:
Thanks, fixed above also.